TodoList/sources/TodoList.Services/Dependency/RegisterTypes.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Practices.Unity;
+using TodoList.Contracts.Dependency;
+using TodoList.Contracts.Services;
+using TodoList.Services.Helpers;
+
+namespace TodoList.Services.Dependency
+{
+    public class RegisterTypes : IBootstrapper
+    {
+        public IUnityContainer RegisterType(IUnityContainer container)
+        {
+            container.RegisterType<IGenerateIdService, GenerateIdService>(new ContainerControlledLifetimeManager());
+            container.RegisterType<ICreateNodeService, CreateNodeService>(new ContainerControlledLifetimeManager());
+            container.RegisterType<ICurrentTimeService, CurrentTimeService>(new ContainerControlledLifetimeManager());
+            container.RegisterType<IUpdateNodeService, UpdateNodeService>(new ContainerControlledLifetimeManager());
```

Comment:
You actually need `HierarchicalLifetimeManager` – read on why [here](https://msdn.microsoft.com/en-us/library/ff660872(v=pandp.20).aspx)